app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/LiveDataFragment.java

```diff
@@ -319,21 +319,20 @@ public int compare(VideoEntity o1,
                                         });
 
                                 // create current category row
-                                ListRow row = new ListRow(new HeaderItem(categoryEntity.getCategoryName()),
-                                        adapter);
+                                ListRow row = new ListRow(categoryEntity.getId(),
+                                        new HeaderItem(categoryEntity.getCategoryName()), adapter);
                                 rows.add(row);
                             }
 
                             mRowsAdapter.setItems(rows, new Comparator<ListRow>() {
                                 @Override
                                 public int compare(ListRow o1, ListRow o2) {
-                                    return o1.getHeaderItem().getName().equals(
-                                            o2.getHeaderItem().getName()) ? 0 : -1;
+                                    return o1.getId() == o2.getId() ? 0 : -1;
                                 }
                             }, new Comparator<ListRow>() {
                                 @Override
                                 public int compare(ListRow o1, ListRow o2) {
-                                    return o1.getHeaderItem().getName().equals(o2.getHeaderItem().getName()) ? 0 : -1;
+                                    return o1.getId() == o2.getId() ? 0 : -1;
```

Comment:
ContentIsSame should compare name, not id

Reply:
Yes, you are right. Sorry for this silly bug.